Lorem Ipsum XR Translator/Assets/Scripts/SpeechToText/SpeechToTextDisplay.cs

```diff
@@ -0,0 +1,177 @@
+        /// </summary>
+        private IEnumerator TranslateAndDisplay(string text)
+        {
+            yield return azureTranslator.Translate(text, "en", new string[] { targetLanguage }, response =>
+            {
+                // Deserialize the root array
+                TranslationResponse[] translationResponses = JsonUtility.FromJson<Wrapper>("{\"array\":" + response + "}").array;
+
+                
+                /// <summary>
+                /// Checks if we got any responses
+                /// </summary>
+                
+                if (translationResponses != null && translationResponses.Length > 0)
+                {
+                    TranslationResponse translationResponse = translationResponses[0];
+
+                    
+                    /// <summary>
+                    /// Checks if the translationResponse contains translations
+                     /// </summary>
+                    if (translationResponse.translations != null && translationResponse.translations.Length > 0)
+                    {
+                        string translatedText = translationResponse.translations[0].text;
+                        DisplayInToolTip(translatedToolTip, translatedText);
+                    }
+                    else
+                    {
+                        Debug.LogError("Failed to extract translation from response.");
+                    }
+                }
+                else
+                {
+                    Debug.LogError("No translation responses received.");
+                }
+            });
+        }
+
+         /// <summary>
+        /// DisplayInToolTip method sets the provided text to the given tooltip and makes the tooltip visible
+        /// </summary>
+        private void DisplayInToolTip(ToolTip toolTip, string text)
+        {
+            if (toolTip != null)
+            {
+                toolTip.ToolTipText = text;
+                toolTip.gameObject.SetActive(true);
+            }
+            else
+            {
+                Debug.LogError("ToolTip is not assigned.");
+            }
+        }
+
+         /// <summary>
+        ///TranslationResponse,TranslationItem and Wrapper these are serializable helper classes used to deserialize the JSON response from the Azure Translator service.
+        /// It requests Azure's Speech to Text service and translation service to recognize speech and then displays the recognized and translated text in the tooltips.
+        /// </summary>
+        [System.Serializable]
+        public class TranslationResponse
+        {
+            public TranslationItem[] translations;
+        }
+
+        [System.Serializable]
+        public class TranslationItem
+        {
+            public string text;
+            public string to;
+        }
+
+        [System.Serializable]
+        public class Wrapper
+        {
+            public TranslationResponse[] array;
+        }
```

Comment:
Nested classes could cause problems. Please move these classes outside of the main class. (They're publicly accessible anyway, so they should be fine if just moved outside of the parent class.

Reply:
I have moved the nested classes outside the scope of main class SpeechToTextDisplay.It should be fine now.
Thank you for the detailed review , it helps a lot